src/main/webapp/profile.js

```diff
@@ -13,15 +13,37 @@
 // limitations under the License.
 
 function onProfileLoad() {
+  autofillForm(getUserIfRegistered());  
+  supplyLogoutLink();
```

Comment:
supplyLogoutLink() should be called first because it redirects the user to the home page if they are not logged in.